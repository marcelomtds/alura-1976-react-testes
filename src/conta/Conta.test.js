import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import Conta from "./Conta";

describe("Component de conta", () => {
  it("O snapshot do component deve permanecer sempre o mesmo", () => {
    const { container } = render(<Conta saldo={20.0} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Exibir o saldo da conta como valor monetário", () => {
    render(<Conta saldo={20.0} />);

    const saldo = screen.getByTestId("saldo-conta");

    expect(saldo.textContent).toMatchSnapshot();
  });

  it("Chama a função de realizar transação quando o botão é clicado", () => {
    const funcaoRealizarTransacao = jest.fn();

    render(<Conta saldo={20.0} realizarTransacao={funcaoRealizarTransacao} />);

    const transacao = screen.getByLabelText("Saque");
    const valor = screen.getByTestId("valor");

    fireEvent.click(transacao, { target: { value: "saque" } });
    fireEvent.change(valor, { target: { value: 10 } });

    fireEvent.click(screen.getByText("Realizar operação"));

    expect(funcaoRealizarTransacao).toHaveBeenCalled();
  });

  it("Não chama a função de realizar transação quando o botão é clicado", () => {
    const funcaoRealizarTransacao = jest.fn();

    render(<Conta saldo={20.0} realizarTransacao={funcaoRealizarTransacao} />);

    fireEvent.click(screen.getByText("Realizar operação"));

    expect(funcaoRealizarTransacao).not.toHaveBeenCalled();
  });
});
