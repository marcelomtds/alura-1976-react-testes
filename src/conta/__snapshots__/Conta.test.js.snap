// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component de conta Exibir o saldo da conta como valor monetário 1`] = `"R$ 20"`;

exports[`Component de conta O snapshot do component deve permanecer sempre o mesmo 1`] = `
<div
  class="Conta-header"
>
  <h2>
    Conta
  </h2>
  <p>
    Saldo: 
    <span
      class="Saldo-valor"
      data-testid="saldo-conta"
    >
      R$ 20
    </span>
  </p>
  <form>
    <div>
      <label>
        Depósito
        <input
          data-testid="transacao"
          name="transacao"
          type="radio"
          value="deposito"
        />
      </label>
    </div>
    <div>
      <label>
        Saque
        <input
          data-testid="transacao"
          name="transacao"
          type="radio"
          value="saque"
        />
      </label>
    </div>
    <label>
      Valor:
    </label>
    <input
      data-testid="valor"
      name="valor"
      type="text"
      value="0"
    />
    <div>
      <button
        disabled=""
        type="submit"
      >
        Realizar operação
      </button>
    </div>
  </form>
</div>
`;
